crates/typst/src/export/pdf/external_graphics_state.rs

```diff
@@ -0,0 +1,37 @@
+use crate::export::pdf::{PdfContext, RefExt};
+use pdf_writer::Finish;
+
+/// A PDF external graphics state.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct ExternalGraphicsState {
+    // In the range 0-255, needs to be divided before being written into the graphics state!
+    pub stroke_opacity: u8,
+    // In the range 0-255, needs to be divided before being written into the graphics state!
+    pub fill_opacity: u8,
+}
```

Comment:
I'm not very happy with that solution yet. It would be nice if we were able to store the opacities directly as floats between 0-1, but then it won't work with the Remapper because f32 doesn't implement `Eq`. Any suggestions?

Reply:
I haven't looked closely yet, but `typst::geom::Scalar` is an f64 that does implement `Eq` at the cost of panicking if it is NaN.